modules/dnn/src/onnx/onnx_importer.cpp

```diff
@@ -1407,6 +1435,49 @@ void ONNXImporter::parseMul(LayerParams& layerParams, const opencv_onnx::NodePro
             //Replace input to Power
             node_proto.set_input(1, powerParams.name);
         }
+
+        const MatShape& broadShape = outShapes[node_proto.input(1)];
+        const MatShape& outShape = outShapes[node_proto.input(0)];
+        const size_t outShapeSize = outShape.size();
+        const size_t diff = outShapeSize - broadShape.size();
+
+        size_t axis = diff;
+        for (; axis < broadShape.size() && broadShape[axis - diff] == 1; ++axis) {}
+
+        size_t endAxis = broadShape.size();
+        for (; endAxis > axis && broadShape[endAxis - 1 - diff] == 1; --endAxis) {}
+
+        int broadAxis = -1;
+        for (size_t i = axis; i < endAxis; ++i)
+        {
+            if (outShape[i] == broadShape[i])
+            {
+                continue;
+            }
+
+            CV_Assert(broadShape[i - diff] == 1 && broadAxis == -1);
+            broadAxis = i;
+        }
+
+        if (broadAxis != -1)
+        {
+            opencv_onnx::NodeProto concat_node_proto = node_proto;
+            const std::string& input1 = concat_node_proto.input(1);
```

Comment:
`String` instead of `std::string` . In 3.4 branch they are different.

Reply:
Prefer to use `std::string` for all branches.